packages/sync-actions/src/product-types-actions.js

```diff
@@ -0,0 +1,205 @@
+  { action: 'changeName', key: 'name' },
+  { action: 'setKey', key: 'key' },
+  { action: 'setDescription', key: 'description' },
+]
+
+export function actionsMapBase(diff, previous, next) {
+  return buildBaseAttributesActions({
+    diff,
+    actions: baseActionsList,
+    oldObj: previous,
+    newObj: next,
+  })
+}
+
+const attributeDefinitionsActionsList = [
+  { action: 'changeLabel', key: 'label' },
+  { action: 'setInputTip', key: 'inputTip' },
+  { actionKey: 'newValue', action: 'changeInputHint', key: 'inputHint' },
+  { action: 'changeIsSearchable', key: 'isSearchable' },
+  {
+    actionKey: 'newValue',
+    action: 'changeAttributeConstraint',
+    key: 'attributeConstraint',
+  },
+]
+
+const getIsAnAttributeDefinitionBaseFieldChange = diff =>
+  diff.label ||
+  diff.inputHint ||
+  diff.inputTip ||
+  diff.attributeConstraint ||
+  diff.isSearchable
+
+function actionsMapEnums(attributeType, attributeDiff, previous, next) {
+  const getAddActionName = type =>
+    type === 'enum' ? 'addPlainEnumValue' : 'addLocalizedEnumValue'
+  const getChangeEnumOrderActionName = type =>
+    type === 'enum'
+      ? 'changePlainEnumValueOrder'
+      : 'changeLocalizedEnumValueOrder'
+  const getChangeEnumLabelActionName = type =>
+    type === 'enum'
+      ? 'changePlainEnumValueLabel'
+      : 'changeLocalizedEnumValueLabel'
+
+  const handler = createBuildArrayActions('values', {
+    [ADD_ACTIONS]: newEnum => ({
+      attributeName: next.name,
+      action: getAddActionName(attributeType),
+      value: newEnum,
+    }),
+    [CHANGE_ACTIONS]: (oldEnum, newEnum) => {
+      const oldEnumInNext = next.values.find(
+        nextEnum => nextEnum.key === oldEnum.key
+      )
+      const changeActions = []
+      if (oldEnumInNext) {
+        if (!deepEqual(oldEnum.label, oldEnumInNext.label)) {
+          changeActions.push({
+            attributeName: next.name,
+            action: getChangeEnumLabelActionName(attributeType),
+            newValue: newEnum,
+          })
+        } else {
+          changeActions.push({
+            attributeName: next.name,
+            action: getChangeEnumOrderActionName(attributeType),
+            value: newEnum,
+          })
+        }
+      } else {
+        changeActions.push({
+          attributeName: next.name,
+          action: 'removeEnumValue',
+          value: oldEnum,
+        })
+        changeActions.push({
+          attributeName: next.name,
+          action: getAddActionName(attributeType),
+          value: newEnum,
+        })
+      }
+      return changeActions
+    },
+    [REMOVE_ACTIONS]: deletedEnum => ({
+      attributeName: next.name,
+      action: 'removeEnumValue',
+      value: deletedEnum,
+    }),
+  })
+  const actions = []
+  const removedKeys = []
+  const newEnumValuesOrder = []
+  flatten(handler(attributeDiff, previous, next)).forEach(action => {
+    if (action.action === 'removeEnumValue') removedKeys.push(action.value.key)
```

Comment:
I think `action.action` looks weird. Could you maybe destructure the parameter to the `forEach` callback

Reply:
no destructs please. this is technically an `updateAction` anyway.